### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by FBruzzesi**
Not super exciting PR, but I found some typos in the codebase fast and easy to fix 😊

Edit: Test are failing with some precision difference in the order of 1e-6, which is the default of `numpy.testing.assert_array_almost_equal`

### CODE REVIEW

File: pomegranate/_utils.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def _cast_as_parameter(value, dtype=None, requires_grad=False):
 
 
 def _update_parameter(value, new_value, inertia=0.0, frozen=None):
-	"""Update a parameters unles.
+	"""Update a parameter unles.
```
Comment (FBruzzesi) [inline pomegranate/_utils.py:81]: I am not sure what the docstring should be here

Comment (jmschrei) [reply pomegranate/_utils.py:81]: It seems perfectly clear. You update a parameter unles.

**Comment by jmschrei** [comment]
Reading this PR is embarrassing.

**Comment by FBruzzesi** [comment]
> Reading this PR is embarrassing.

It happens to everyone, no need to feel embarrassed! If I may suggest a like-able tool to semi automate this process: [typos](https://github.com/crate-ci/typos) is (very) fast, flexible and customizable.

**Pull request merged**